code/openai_synthesize.py

```diff
@@ -1,23 +1,36 @@
 import argparse
 import json
 from openai import OpenAI
+from openrouter_client import OpenRouterClient
 from prompt_templates import instruction_template, knowledge_template, npc_template, math_template
 from datasets import load_dataset
 from tqdm import tqdm
 
 system_prompt = '''You are a helpful assistant.'''
-client = OpenAI()   # set up your config/env/api for calling openai models
 
-def get_response(user_prompt):
-    completion = client.chat.completions.create(
-        model="gpt-4o",
-        temperature=0.7,
-        messages=[
-            {"role": "system", "content":  f"{system_prompt}"},
-            {"role": "user", "content": f"{user_prompt}"}
-        ]
-    )
-    return completion.choices[0].message.content
+def get_response(user_prompt, use_openrouter=False):
+    if use_openrouter:
+        client = OpenRouterClient()
+        completion = client.chat.create(
+            model="openai/gpt-4",  # Use an appropriate model supported by OpenRouter
```

Comment:
We should ask it to implement an extra args to pass down to this function, so that user can do --model="etc"

Also, should prob keep `gpt4` here TBH